### PR TITLE
Resolve spec inside  and json_array and json_object

### PR DESCRIPTION
**Pull request by aaclan-ebi**

### CODE REVIEW

File: json_converter/json_mapper.py [hunk 1]
```diff
@@ -114,15 +113,27 @@ def _apply_field_spec(node: DataNode, spec: list):
                 field_value = operation(*args)
         return field_value
 
-    @staticmethod
-    def _get_object_literal(spec):
-        if len(spec) != 2:
+    def _get_object_literal(self, spec):
+        if len(spec) < 2 or len(spec) > 3:
             raise UnreadableSpecification('Expecting exactly 1 JSON literal value.')
```
Comment (ke4) [inline json_converter/json_mapper.py]: I think the message should be: `Expecting 2 or 3 JSON literal value.`

**Review by ke4 — CHANGES_REQUESTED** [review]
Excellent work! 👍 💯 
Request for 2 tiny modifications:
- Please change the error message according to the condition change
- bump the version in `setup.py` to be able to deploy `pipy` with a new version. I would bump it to 0.4.0.
Many thanks

**Comment by aaclan-ebi** [comment]
@ke4 I'm still not sure if there's a better way to provide the use case you needed. I'll let you merge and release this to PyPI. I've added other devs to check if they have thoughts on this.

**Comment by MightyAx** [comment]
I was worried that this change would break my existing uses of the $object literal to use literal json that I did not want to be converted but having tested the changes, that is not the case 👍

@aaclan-ebi, @ke4: The change definitely allows you to build structure (the key/value attributes) around mapped attributes.
Can this code also be used for unmapped attributes? This would simplify some code where I currently must do this manually.

```python
spec = {
  'A': ['a'],
  'others': ['$unmapped', map_to, {
    'KEY': '$key',
    'VALUE': '$value'
  }] # This functionality doesn't exist but something like it would be ideal.
}

input_object = {
  'a': 1
  'b': 2
  'c': 3
}

result = {
  'A': 1
  'others': [
    {
      'KEY': 'b',
      'VALUE': 2
    },
    {
      'KEY': 'c',
      'VALUE': 3
    }
  ]
}
```

Currently to achieve the above I would keep a list of the mapped keys next to spec, pop them from the dictionary and map the 'others' object separately. Is that still the best way to do this?

```python
remove_keys = ['a']

others = copy(input_object)
for key in remove_keys:
  others.pop(key)
```
so now I have this others object:
```
others = {
  'b': 2
  'c': 3
}
```
what is the syntax to make this without referring to the b & c keys?:
```
[
  {
    'KEY': 'b',
    'VALUE': 2
  },
  {
    'KEY': 'c',
    'VALUE': 3
  }
]
```
or perhaps this, whichever is easier:
```
{
  'others': [
    {
      'KEY': 'b',
      'VALUE': 2
    },
    {
      'KEY': 'c',
      'VALUE': 3
    }
  ]
}
```

File: README.md [hunk 2]
```diff
@@ -377,3 +378,70 @@ The `json_array` method treats argument list as a list of JSON objects. For exam
...
+            },
+            {
+                'key': ['', default_to, 'address'],
+                'value': ['address']
+
+            },
+        ]
+
+result = json_mapper.map({
+            'metadata': ['$array', values, True]
```
Comment (ke4) [inline README.md]: Please, replace the metadata key with attributes, as that is the key in the resulting JSON.
```suggestion
            'attributes': ['$array', values, True]
```

**Review by ke4 — APPROVED** [review]

**Pull request merged**